File: frontend/bff/src/application/orders/commands/register-order.command.ts
import type { OrderTermsInput } from './order-terms.input.js';
import type { RequestContext } from '../../../cross-cutting/context/request-context.js';

export interface RegisterOrderCommand {
  context: RequestContext;
  /** Nulo = boleta sem mandato (exige justificativa nos termos). */
  mandateId: string | null;
  counterpartyId: string;
  terms: OrderTermsInput;
}

File: backend/core-service/src/Fix.Application/UseCases/Orders/Commands/RegisterOrderCommand.cs
using Fix.Application.Abstractions.Authorization;
using Fix.Application.Abstractions.Context;
using Fix.Application.Abstractions.Messaging;
using Fix.Application.Orders.Dtos;
using Fix.Domain.AggregateRoots.Roles;

namespace Fix.Application.Orders.Commands;

[RequireRole(RoleCodes.CreateOrder)]
public sealed record RegisterOrderCommand(
    Guid UserId,
    Guid OrganizationId,
    /// <summary>Nulo = boleta sem mandato (desvio: exige justificativa e fica exposto).</summary>
    Guid? MandateId,
    Guid CounterpartyId,
    OrderTermsInput Terms)
    : ICommand<OrderDto>, IOrganizationRequest;

